ckb_consts.h

```diff
@@ -21,6 +21,7 @@
 #define CKB_INDEX_OUT_OF_BOUND 1
 #define CKB_ITEM_MISSING 2
 #define CKB_LENGTH_NOT_ENOUGH 3
+#define CKB_USER_ERROR 4
```

Comment:
`CKB_USER_ERROR ` is too general for the purpose here, from my observation maybe we can rename it to `CKB_INVALID_DATA_ERROR`